protobuf/message.proto

```diff
@@ -0,0 +1,172 @@
+syntax = "proto3";
+
+package protobuf;
+
+message SyncContact {
+  // The public key of the contact to be synced
+  string public_key = 1;
+  // The profile image of the contact
+  string profile_image = 2;
+  // The ens_name of the contact
+  string ens_name = 3;
+  // The system tags for a given contact
+  repeated SystemTags system_tags = 4;
+
+  enum SystemTags {
+    ADDED_BY_US = 0;
+    BLOCKED = 1;
+    ADDED_BY_THEM = 2;
+  }
+}
+
+message SyncPublicChat {
+  // The chat_id of the chat to be synced
+  string chat_id = 1;
+}
+
+message SyncAccount {
+  // Our profile image
+  string profile_image = 1;
+  // The time we last updated the account
+  uint64 last_updated = 2;
+}
+
+message SyncInstallation {
+  // The contacts to be synced
+  repeated SyncContact contacts = 1;
+  // The public chats to be synced
+  repeated SyncPublicChat public_chats = 2;
+  // Our own account
+  repeated SyncAccount account = 3;
+}
+
+message PairInstallation {
+  // The user-set device name
+  string device_name = 1;
+  // The installation-id of the device
+  string installation_id = 2;
+  // The type of device
+  DeviceType device_type = 3;
+  enum DeviceType {
+    ANDROID = 0;
```

Comment:
often it pays off to keep 0 as 'UNKNOWN', makes it easier to deal with backwards compat when adding a new value

Reply:
in particular, it helps because it forces current clients to define behaviour for what to do when there's an unrecognised enum value, meaning they will hopefully behave sanely in the future

Reply:
Makes sense, thanks!